### PR TITLE
Fix compilation with recent LLVM

### PR DESCRIPTION
**Pull request by thewilsonator**
There are still a bunch of deprecations with `CreateLoad` that need to be addressed at some point.

This is based off of #3720 because I can't build without that.

### CODE REVIEW

File: driver/codegenerator.cpp [hunk 1]
```diff
@@ -263,9 +263,9 @@ void CodeGenerator::writeAndFreeLLModule(const char *filename) {
   Version.append(global.ldc_version.ptr, global.ldc_version.length);
   llvm::Metadata *IdentNode[] = {llvm::MDString::get(ir_->context(), Version)};
   IdentMetadata->addOperand(llvm::MDNode::get(ir_->context(), IdentNode));
-
+#if LDC_LLVM_VER < 1300
   context_.setInlineAsmDiagnosticHandler(inlineAsmDiagnosticHandler, ir_);
-
+#endif
```
Comment (thewilsonator) [inline driver/codegenerator.cpp:268]: This seems to have been removed. I'm not sure what, if anything has replaced it.

Comment (kubo39) [reply driver/codegenerator.cpp:268]: looks like unified in llvm::DiagnoseInfoSrcMgr.
https://reviews.llvm.org/rG5de2d189e6ad466a1f0616195e8c524a4eb3cbc0

File: gen/coverage.cpp [hunk 2]
```diff
@@ -38,6 +38,9 @@ void emitCoverageLinecountInc(const Loc &loc) {
 
   // Do an atomic increment, so this works when multiple threads are executed.
   gIR->ir->CreateAtomicRMW(llvm::AtomicRMWInst::Add, ptr, DtoConstUint(1),
+#if LDC_LLVM_VER >= 1300
+                           llvm::MaybeAlign(),
```
Comment (JohanEngelen) [inline gen/coverage.cpp:42]: this can be explicit 4-byte (uint) alignment

**Comment by thewilsonator** [comment]
superseded by #3822

**Pull request closed without merging**